<Window x:Class="WpfApplication1.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:local="clr-namespace:WpfApplication1"
        mc:Ignorable="d"
        Title="MainWindow" Height="350" Width="525">
    <Grid x:Name="LayoutRoot" Background="White">
        <StackPanel Orientation="Vertical">
            <Button Content="Receive FaultException (without try/catch around the call)" Foreground="White" Background="#FFE44D26" Click="ButtonReceiveFaultExceptionWithoutTryCatch" HorizontalAlignment="Right" Margin="0,5,0,0"/>
            <Button Content="Receive FaultException (with try/catch)" Foreground="White" Background="#FFE44D26" Click="ButtonReceiveFaultExceptionWithTryCatch" HorizontalAlignment="Right" Margin="0,5,0,0"/>
            <Button Content="Receive Exception (without try/catch around the call)" Foreground="White" Background="#FFE44D26" Click="ButtonReceiveServerErrorWithoutTryCatch" HorizontalAlignment="Right" Margin="0,5,0,0"/>
            <Button Content="Receive Exception (with try/catch)" Foreground="White" Background="#FFE44D26" Click="ButtonReceiveServerErrorWithTryCatch" HorizontalAlignment="Right" Margin="0,5,0,0"/>
            <Button Content="Receive FaultException (event-based)" Foreground="White" Background="#FFE44D26" Click="ButtonReceiveFaultExceptionEventAsync" HorizontalAlignment="Right" Margin="0,5,0,0"/>
            <Button Content="Receive Exception (event-based)" Foreground="White" Background="#FFE44D26" Click="ButtonReceiveServerErrorEventAsync" HorizontalAlignment="Right" Margin="0,5,0,0"/>
        </StackPanel>
    </Grid>
</Window>
